### ISSUE TITLE
MD034/no-bare-urls false positive since v0.7.0

### ISSUE THREAD

**Issue by MichaIng** [DavidAnson/markdownlint-cli2#150]
This code produces an error since v0.7.0:
```md
[[text] text](https://some.url.org)
```
Or are there Markdown parsers which do not support brackets within the link text brackets? In case, is there a proper way to escape them?

**Comment by DavidAnson** [comment]
Per the specification, if "text" were a valid shortcut link, then it would bind more tightly than the outer link and this error message would be correct.

Example here: https://dlaa.me/markdownlint/#%25m%23%20Issue%20150%0A%0A%5B%5Btext%5D%20text%5D(https%3A%2F%2Fsome.url.org)%0A%0A%5Btext%5D%3A%20https%3A%2F%2Fexample.com%0A

In this case, "text" is not a valid shortcut link, but because the linter assumes all shortcut links are valid even if undefined (needed for other reasons), it produces this misleading output.

**Comment by DavidAnson** [comment]
As you suggest, it is perfectly valid to escape the inner brackets like so to avoid the warning: https://dlaa.me/markdownlint/#%25m%23%20Issue%20150%0A%0A%5B%5C%5Btext%5C%5D%20text%5D(https%3A%2F%2Fsome.url.org)%0A
